src/Nethermind/Nethermind.Db/DbExtensions.cs

```diff
@@ -200,13 +200,50 @@ public static void Delete(this IDb db, long key)
                     item = decoder.Decode(data.AsRlpStream(), RlpBehaviors.AllowExtraData);
                 }
             }
-            
+
             if (shouldCache && cache != null && item != null)
             {
                 cache.Set(key, item);
             }
 
             return item;
         }
+
+        public static bool? LoadBooleanFromDb(this IDb db, int key)
+        {
+            try
+            {
+                if (db.KeyExists(key))
+                {
+                    byte[]? hashFromDb = db.Get(key);
+                    RlpStream stream = new(hashFromDb!);
+                    return stream.DecodeBool();
+                }
+            }
+            catch (RlpException)
+            {
```

Comment:
maybe logger?

Reply:
Extension method of an interface, we got no access to private field, I'd have to expose a logger to the interface to have log.